api/turing/service/router_deployment_service.go

```diff
@@ -626,15 +666,15 @@ func deleteKnServices(
 		errCh <- err
 	}
 
-	return updateKnServices(ctx, services, deleteFunc, eventsCh)
+	return updateResources(ctx, services, deleteFunc, eventsCh)
 }
 
-// updateKnServices is a helper method for deployment / deletion of services that runs the
+// updateResources is a helper method for deployment / deletion of resources that runs the
 // given update function on the given services simultaneously and waits for a response,
 // within the supplied timeout.
-func updateKnServices(ctx context.Context, services []*cluster.KnativeService,
-	updateFunc uFunc, eventsCh *EventChannel) error {
-
+func updateResources[V *cluster.KnativeService | *cluster.PodDisruptionBudget](ctx context.Context, services []V,
```

Comment:
I'm refactoring updateKnServices to be updateResources that use generic so it can accept both KnativeServie and PodDisruptionBudget. Let me know what do you think about this. @krithika369